pkg/ref/expr.go

```diff
@@ -146,7 +153,28 @@ func (r *resolver) getSecret(secret *corev1.Secret, namespace, name string, vali
 			}, name)
 		}
 	}
-	return r.req.Get(r.ctx, router.Key(namespace, name), secret)
+	if err := r.req.Get(r.ctx, router.Key(namespace, name), secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Try finding by public name
+			secretList := &corev1.SecretList{}
+			if err := r.req.List(r.ctx, secretList, &kclient.ListOptions{
+				LabelSelector: klabels.SelectorFromSet(map[string]string{
+					labels.AcornPublicName: name,
+				}),
+				Namespace: namespace,
+			}); err != nil {
+				return err
+			}
+
+			if len(secretList.Items) == 1 {
+				secretList.Items[0].DeepCopyInto(secret)
+				return nil
+			}
+		}
+		return err
+	}
+
+	return nil
```

Comment:
nit:
```suggestion
	if err := r.req.Get(r.ctx, router.Key(namespace, name), secret); !apierrors.IsNotFound(err) {
		return err
	}

	// Try finding by public name
	secretList := &corev1.SecretList{}
	if err := r.req.List(r.ctx, secretList, &kclient.ListOptions{
		LabelSelector: klabels.SelectorFromSet(map[string]string{
			labels.AcornPublicName: name,
		}),
		Namespace: namespace,
	}); err != nil {
		return err
	}

	if len(secretList.Items) == 1 {
		secretList.Items[0].DeepCopyInto(secret)
		return nil
	}

	return nil
```

Reply:
I structured this in an intentional way to return the original `NotFound` error if there are more or less than 1 item in the secretList. I can restructure it similarly to your suggestion here (it would involve preserving the `err` from `r.req.Get` at the beginning and returning it instead of `nil` at the end of the function), but imo the code is a little bit tougher to follow if structured that way, even though it would be less nested.

Reply:
Got it. Leave it as-is. I like it.